com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -681,63 +954,141 @@ void ResetSensors()
+        /// force to move an agent around the environment, you can arbitrarily pick
+        /// three values in the action array to use as the force components. During
+        /// training, the agent’s  policy learns to set those particular elements of
+        /// the array to maximize the training rewards the agent receives. (Of course,
+        /// if you implement a <seealso cref="Heuristic"/> function, it must use the same
+        /// elements of the action array it creates for the same purpose since there
+        /// is no learning involved.)
+        /// 
+        /// Actions for an agent can be either *Continuous* or *Discrete*. Specify which
+        /// type of action space an agent uses, along with the size of the action array,
+        /// in the <see cref="BrainParameters"/> of the agent’s associated
+        /// <see cref="BehaviorParameters"/> component. 
+        /// 
+        /// When an agent uses the continuous action space, the values in the action
+        /// array are floating point numbers. You should clamp the values to the range,
+        /// -1..1, to increase numerical stability during training.
+        /// 
+        /// When an agent uses the discrete action space, the values in the action array
+        /// are integers that each represent a specific, discrete action. For example,
+        /// you could define a set of discrete actions such as:
+        ///
+        /// <code>
+        /// 0 = Do nothing
+        /// 1 = Move one space left
+        /// 2 = Move one space right
+        /// 3 = Move one space up
+        /// 4 = Move one space down
+        /// </code>
+        /// 
+        /// When making a decision, the agent picks one of the five actions and puts the
+        /// corresponding integer value in the action vector. For example, if the agent
+        /// decided to move left, the action vector parameter would contain an array with
+        /// a single element with the value 1.
+        ///
+        /// You can define multiple sets, or branches, of discrete actions to allow an
+        /// agent to perform simultaneous, independent actions. For example, you could
+        /// use one branch for movement and another branch for throwing a ball left, right,
+        /// up, or down, to allow the agent to do both in the same step. 
+        /// 
+        /// The action vector of a discrete action space contains one element for each
+        /// branch. The value of each element is the integer representing the chosen
+        /// action for that branch. The agent always chooses one action for each
+        /// branch.
+        /// 
+        /// When you use the discrete action space, you can prevent the training process
+        /// or the neural network model from choosing specific actions in a step by
+        /// implementing the <see cref="CollectDiscreteActionMasks(DiscreteActionMasker)"/>
+        /// function. For example, if your agent was learning to play chess, you could
```

Comment:
I would advise against mentioning chess here as I think ML-Agents would not be able to solve chess. You were using an example of a gridworld where the agent moves up right left down, maybe stick to this example.